torch/_inductor/config.py

```diff
@@ -463,6 +463,11 @@ def decide_compile_threads():
 # decompose some memory bound matmul/bmm to mul
 decompose_mem_bound_mm: bool = False
 
+# assume_aligned_inputs means that we assume that inputs will be aligned; we generate
+# code using this assumption, and clone tensors before use if they aren't aligned.
+# In the common case, most inputs will be aligned.
+assume_aligned_inputs: bool = True
```

Comment:
I'm a little confused by the naming.  If we are cloning unaligned tensors, aren't we _forcing_ the tensors to be aligned, not _assuming_ they are going to be aligned?  Or is this an inductor vs rest of the system perspective thing?

Reply:
Whoops, didn't see this comment until now.

I guess my thought process was that, when this flag is turned on, inductor generates triton code that **assumes its inputs will be aligned**.

But I can see why this is confusing - do you have a suggestion for a better name?

Reply:
or, put another way, assume_aligned_inputs=True -> optimize for the case where inputs are aligned, assume_aligned_inputs=False -> optimize for the case where inputs are unaligned